packages/mco/components/data-policies/create-dr-policy/create-dr-policy.tsx

```diff
@@ -138,31 +230,131 @@ type SyncScheduleProps = {
   setSchedule: React.Dispatch<React.SetStateAction<string>>;
 };
 
-export const CreateDRPolicy: React.FC<{}> = ({}) => {
+type ReRouteResourceProps = {
+  history: RouteComponentProps['history'];
+  match: Match<{ url: string }>;
+};
+
+export const CreateDRPolicy: React.FC<ReRouteResourceProps> = ({match, history}) => {
   const { t } = useTranslation('plugin__odf-console');
-  const history = useHistory();
+  const { url }  = match;
+  
   const [policyName, setPolicyName] = React.useState('');
-  const [replication] = React.useState('');
-  const [selectedClusters, setSelectedClusters] = React.useState<Cluster[]>([]);
+  const [replication, setReplication] = React.useState('');
+  const [manageS3Profile, setManageS3Profile] = React.useState(false);
   const [syncTime, setSyncTime] = React.useState('5m');
+  const [isODFDetected, setODFDetected] = React.useState(false);
+  const [selectedClusters, setSelectedClusters] = React.useState<ManagedClusterMapping>({});
+  const [isReplicationOpen, setReplicationOpen] = React.useState(false);
+  const [isReplicationInputManual, setIsReplicationInputManual] = React.useState(false);
+  const [errorMessage, setError] = React.useState("");
+  const [mirrorPeers] = useK8sWatchResource<MirrorPeerKind[]>({
+    kind: referenceForModel(MirrorPeerModel),
+    isList: true,
+    namespaced: false,
+  });
+
+  const clustersData =  React.useMemo(() => Object.values(selectedClusters) ?? [], [selectedClusters]);
+
+  React.useEffect(() => {
+    if (clustersData?.length === 2) {
+      // ODF detection
+      setODFDetected(clustersData?.every((cluster) => (cluster?.storageSystem !== "" && cluster?.isValidODFVersion)));
```

Comment:
```suggestion
      setODFDetected(clustersData.every((cluster) => (cluster?.storageSystem !== "" && cluster?.isValidODFVersion)));
```
Isn't `clusterData` a guaranteed empty array at the very least. I don't think optional chaining is required.